### PR TITLE
Update GitHub action

### PR DESCRIPTION
**Pull request by zengyan-amazon**
*Issue #, if available:*

*Description of changes:* build plugin in github actions
tested:  https://github.com/zengyan-amazon/security-kibana-plugin-1/actions/runs/267759964

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    - name: Checkout Kibana
+      uses: actions/checkout@v2
+      with:
+        path: kibana
+        repository: elastic/kibana
+        fetch-depth: 0
+    - name: Use release
+      run: |
+        cd ./kibana
+        git checkout tags/v7.9.1 -b v7.9.1
```
Comment (bandinib-amzn) [inline .github/workflows/cd.yml:22]: can we make version `v7.9.1` configurable? Does it need to be hardcoded?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -12,17 +12,16 @@ jobs:
...
     - name: Checkout Kibana
       uses: actions/checkout@v2
       with:
         path: kibana
         repository: elastic/kibana
+        fetch-depth: 0
+    - name: Use release
+      run: |
+        cd ./kibana
+        git checkout tags/v7.9.1 -b v7.9.1
```
Comment (bandinib-amzn) [inline .github/workflows/ci.yml:24]: same comment as above

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -45,16 +44,30 @@ jobs:
...
+        node build_tools/build_plugin.js
+        artifact_path=`ls $(pwd)/build/opendistro_security_kibana_plugin-*.zip`
+        artifact_name=`basename $artifact_path`
+        echo ::set-env name=ARTIFACT_PATH::$artifact_path
+        echo ::set-env name=ARTIFACT_NAME::$artifact_name
+    - name: Upload Workflow Artifacts
+      uses: actions/upload-artifact@v1
+      with:
+        name: ${{ env.ARTIFACT_NAME }}
+        path: ${{ env.ARTIFACT_PATH }}
```
Comment (yuxitang-amzn) [inline .github/workflows/ci.yml:73]: New line at the end

File: build_tools/build_plugin.js [hunk 4]
```diff
@@ -72,6 +72,11 @@ async function build() {
   console.log('Creating plugin zip bundle...');
   try {
     await createPackage.createPackage(plugin, buildTarget, buildVersion);
+    // rename to opendistro_security_kiban_plugin-{version}.zip to match legacy plugin name
```
Comment (tianleh) [inline build_tools/build_plugin.js:75]: minor typo: kibana

File: .github/workflows/cd.yml (hunk 1 above)
Comment (zengyan-amazon) [reply .github/workflows/cd.yml:22]: yeah, we should be able to get this out of `kibana.json` file, I will fix it in future PR.

File: .github/workflows/cd.yml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    - name: Get node and yarn versions
+      id: versions
+      run: |
+        echo "::set-output name=node_version::$(cat ./kibana/.node-version)"
+        echo "::set-output name=yarn_version::$(cat ./kibana/package.json | jq '.engines.yarn' | tr -d '"')"
```
Comment (yuxitang-amzn) [inline .github/workflows/cd.yml:27]: `jq -r '.engines.yarn' ./kibana/package.json` should be equivalent to  `cat ./kibana/package.json | jq '.engines.yarn' | tr -d '"'` in this case

**Review by bandinib-amzn — APPROVED** [review]

**Review by tianleh — APPROVED** [review]
Follow ups are needed for improvement.

**Pull request merged**